src/main/java/unrc/dose/Proposition.java

```diff
@@ -1,18 +1,417 @@
+        return this.getInteger("user_id");
+    }
+
+    /**
+     *
+     * @return id of challenge
+     */
+    public int getChallengeId() {
+        return this.getInteger("challenge_id");
+    }
+
+    /**
+     *
+     * @return code of the proposed solution
+     */
+    public String getSource() {
+        return this.getString("source");
+    }
+
+    /**
+     *
+     * @return status of the proposed solution
+     */
+    public Boolean getIsSolution() {
+        return this.getBoolean("isSolution");
+    }
+
+    /**
+     *
+     * @return distance obtained from the proposed solution
+     */
+    public int getDistance() {
+        return this.getInteger("distance");
+    }
+
+    /**
+     *
+     * @return amount of tests passed by the proposed solution
+     */
+    public int getCantTestPassed() {
+        return this.getInteger("cantTestPassed");
+    }
+
+    /**
+     * Sets a user id.
+     * @param userId reference new user id for the proposition
+     */
+    public void setUserId(final int userId) {
+        this.set("user_id", userId);
+    }
+
+    /**
+     * Sets a challenge id.
+     * @param challengeId reference new id challenge for the proposition
+     */
+    public void setChallengeId(final int challengeId) {
+        this.set("challenge_id", challengeId);
+    }
+
+    /**
+     * Sets a source's.
+     * @param source reference new source for the proposition
+     */
+    public void setSource(final String source) {
+        if (source == null) {
+            throw new IllegalArgumentException();
+        }
+        this.set("source", source);
+    }
+
+    /**
+     * Set isSolution value.
+     * @param isSolution reference new value
+     * for the isSolution in the proposition
+     */
+    public void setIsSolution(final boolean isSolution) {
+        this.set("isSolution", isSolution);
+    }
+
+    /**
+     * Set distance value calculated previus.
+     * @param distance reference new distance in the proposition.
+     */
+    public void setDistance(final int distance) {
+        this.set("distance", distance);
+    }
+
+    /**
+     * Set cantTestPassed value.
+     * @param testPassed reference the new amount of tests passed
+     */
+    public void setCantTestPassed(final int testPassed) {
+        this.set("cantTestPassed", testPassed);
+    }
+
+    /**
+     * Method for creates new proposition in the database.
+     * @param userId reference the user for the new proposition.
+     * @param challengeId reference the challenge for the new proposition.
+     * @return the proposition created in the database.
+     */
+    public static Proposition newProposition(
+        final int userId,
+        final int challengeId) {
+
+        Challenge currentChallenge = Challenge.findById(challengeId);
+        User currentUser = User.findById(userId);
+
+        if (currentChallenge == null || currentUser == null) {
+            throw new IllegalArgumentException();
+        }
```

Comment:
The exception should include a proper description, using `IllegalArgumentException` should be replaced in this case by a `ChallengeNotFoundException` and `UserNotFoundException` respectively.